plugins/modules/dcnm_template.py

```diff
@@ -334,11 +334,19 @@ def dcnm_template_validate_template(self, template):
             # DATA may have multiple dicts with different reports. Check all reports and ignore warnings.
             # If there are errors, take it as validation failure
 
-            for d in resp["DATA"]:
-                if d["reportItemType"].lower() == "error":
+            # resp['DATA'] may be a list in case of templates with no parameters. But for templates
+            # with parameters resp['DATA'] will be a dict directly with 'status' as 'Template Validation Successful'
+            if isinstance(resp['DATA'], list):
+                for d in resp["DATA"]:
+                    if d.get("reportItemType", ' ').lower() == "error":
+                        self.result["response"].append(resp)
+                        return 0
+                return resp["RETURN_CODE"]
+            elif isinstance(resp['DATA'], dict):
```

Comment:
Should you have a final `else` block in the event that the `resp['DATA']` is something we don't recognize to generate a meaningful/helpful failure?